spec/lib/ruby_smb/server/share/provider/virtual_disk_spec.rb

```diff
@@ -0,0 +1,122 @@
+RSpec.describe RubySMB::Server::Share::Provider::VirtualDisk do
+  let(:name) { 'vdisk' }
+  subject(:virtual_disk) { described_class.new(name) }
+
+  it { is_expected.to respond_to :name }
+  it { is_expected.to respond_to :path }
+  it { is_expected.to respond_to :type }
+  context 'it provides the basic Hash methods' do
+    it { is_expected.to respond_to :[] }
+    it { is_expected.to respond_to :each }
+    it { is_expected.to respond_to :each_key }
+    it { is_expected.to respond_to :each_value }
+    it { is_expected.to respond_to :keys }
+    it { is_expected.to respond_to :length }
+    it { is_expected.to respond_to :values }
+  end
+
+  describe '#add' do
+    let(:path) { '/path' }
+
+    it 'does not throw an exception if the stat object is not a file' do
+      stat = RubySMB::Server::Share::Provider::VirtualDisk::VirtualStat.new(file?: false)
```

Comment:
Not a blocker, just a heads up, `described_class` can be used like this:
```suggestion
      stat = described_class::VirtualStat.new(file?: false)
```